packages/dialect-react-sdk/hooks/useAddresses.ts

```diff
@@ -0,0 +1,398 @@
+import type {
+  Address,
+  DappAddress,
+  AddressType,
+  DialectSdkError,
+} from '@dialectlabs/sdk';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import useSWR from 'swr';
+import { EMPTY_OBJ } from '../utils';
+import {
+  WALLET_ADDRESSES_CACHE_KEY_FN,
+  WALLET_DAPP_ADDRESSES_CACHE_KEY_FN,
+} from './internal/swrCache';
+import useDialectDapp from './useDialectDapp';
+import useDialectSdk from './useDialectSdk';
+import useDialectWallet from './useDialectWallet';
+
+interface CreateUpdateParams {
+  type: AddressType;
+  value: string;
+}
+
+interface DeleteParams {
+  type: AddressType;
+}
+
+interface ToggleParams {
+  type: AddressType;
+  enabled: boolean;
+}
+
+interface VerifyParams {
+  type: AddressType;
+  code: string;
+}
+interface ResendParams {
+  type: AddressType;
+}
+
+type AddressActionParams =
+  | CreateUpdateParams
+  | DeleteParams
+  | ToggleParams
+  | VerifyParams
+  | ResendParams;
+
+type AddressEnriched = Address & {
+  enabled: boolean;
+  dappAddress?: DappAddress;
+};
+interface UseAddressesValue {
+  addresses: Record<AddressType, AddressEnriched> | Record<string, never>;
+  create: (params: CreateUpdateParams) => Promise<void>;
+  update: (params: CreateUpdateParams) => Promise<void>;
+  delete: (params: DeleteParams) => Promise<void>;
+  toggle: (params: ToggleParams) => Promise<void>;
+
+  verify: (params: VerifyParams) => Promise<void>;
+  resend: (params: ResendParams) => Promise<void>;
+
+  isCreatingAddress: boolean;
+  isUpdatingAddress: boolean;
+  isDeletingAddress: boolean;
+  isSendingCode: boolean;
+  isVerifyingCode: boolean;
+
+  isFetching: boolean;
+  errorFetching: DialectSdkError | null;
+}
+
+interface UseAddressesParams {
+  refreshInterval?: number;
+}
+
+function useAddresses({
+  refreshInterval,
+}: UseAddressesParams = EMPTY_OBJ): UseAddressesValue {
+  const { wallet: walletsApi } = useDialectSdk();
+  const { dappAddress: dappPublicKey } = useDialectDapp();
+  const { connected: isWalletConnected } = useDialectWallet();
+  const [isCreatingAddress, setCreatingAddress] = useState(false);
```

Comment:
hey! thanks for the proposal, I don't want to merge all this in one 'cause it could produce unexpected behaviour. E.g. while one address is deleting, another one could be created but the addressState would be set to null in success handler